saltproc/openmc_depcode.py

```diff
@@ -0,0 +1,326 @@
+
+from pyne import nucname as pyname
+from pyne import serpent
+import openmc
+
+from saltproc import Materialflow
+from saltproc.abc import Depcode
+
+class OpenMCDepcode(Depcode):
+    """Class contains information about input, output, geometry, and
+    template files for running OpenMC depletion simulations.
+    Also contains neutrons population, active, and inactive cycles.
+    Contains methods to read template and output files,
+    write new input files for OpenMC.
+
+    Attributes
+    ----------
+    param : dict of str to type
+        Holds depletion step parameter information. Parameter names are keys
+        and parameter values are values.
+    sim_info : dict of str to type
+        Holds simulation settings information. Setting names are keys
+        and setting values are values.
+    iter_inputfile : dict of str to str
+        Paths to OpenMC input files for OpenMC rerunning.
+    iter_matfile : str
+        Path to iterative, rewritable material file for OpenMC
+        rerunning. This file is modified during the simulation.
+
+
+
+    """
+
+    def __init__(self,
+                 exec_path="openmc_deplete.py",
+                 template_input_file_path={"geometry": "./geometry.xml",
+                                           "settings": "./settings.xml",
+                                           "chain_file": "./chain_simple.xml"},
+                 geo_files=None,
+                 npop=50,
+                 active_cycles=20,
+                 inactive_cycles=20):
+        """Initializes the OpenMCDepcode object.
+
+           Parameters
+           ----------
+           exec_path : str
+               Path to OpenMC depletion script.
+           template_input_file_path : dict of str to str
+               Path to user input files (``.xml`` file for geometry,
+               material, and settings) for OpenMC. File type as strings
+               are keys (e.g. 'geometry', 'settings', 'material'), and
+               file path as strings are values.
+           geo_files : str or list, optional
+               Path to file that contains the reactor geometry.
+               List of `str` if reactivity control by
+               switching geometry is `On` or just `str` otherwise.
+           npop : int, optional
+               Size of neutron population per cycle for Monte Carlo.
+           active_cycles : int, optional
+               Number of active cycles.
+           inactive_cycles : int, optional
+               Number of inactive cycles.
+
+        """
+        super().__init__("openmc",
+                         exec_path,
+                         template_input_file_path,
+                         geo_files,
+                         npop,
+                         active_cycles,
+                         inactive_cycles)
+        self.iter_inputfile = {'geometry': './geometry.xml',
+                               'settings': './settings.xml'},
+        self.iter_matfile = './materials.xml'
```

Comment:
Do you mean to leave these paths hardcoded? Wouldn't you want to read in the values of the template_input_file dictionary?

Reply:
I agree with this, it seems restrictive to have these hardcoded.

Reply:
They are hardcoded OpenMC by default expects that the settings, geometry, and material files are named `settings.xml`, `geometry.xml` `materials.xml` respectively. They are also hardcoded because they are modified instead of the user's original OpenMC input files. The user has no need to specify a name for them in the first place.

Reply:
So the user can specify files that they created from any location, then the code will copy them to have these names and be in this location. Then the code continuously calls these copied files and edits them as needed? 

I get hardcoding the names, if that is what OpenMC requires. But would the file path ever change?

Reply:
> So the user can specify files that they created from any location, then the code will copy them to have these names and be in this location. Then the code continuously calls these copied files and edits them as needed?

Correct

> I get hardcoding the names, if that is what OpenMC requires. But would the file path ever change?

The file path will only change if there's a change to OpenMC that requires us to change it as such.